iroh/src/magicsock/transports.rs

```diff
@@ -0,0 +1,432 @@
+use std::{
+    io::{self, IoSliceMut},
+    net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr, SocketAddrV4, SocketAddrV6},
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+use iroh_base::{NodeId, RelayUrl};
+use n0_watcher::Watcher;
+#[cfg(not(wasm_browser))]
+use netwatch::UdpSocket;
+use relay::RelayDatagramSendChannelSender;
+use tracing::{trace, warn};
+
+#[cfg(not(wasm_browser))]
+mod ip;
+mod relay;
+
+#[cfg(not(wasm_browser))]
+pub(crate) use self::ip::IpTransport;
+pub(crate) use self::relay::{RelayActorConfig, RelayTransport};
+use super::NetInfo;
+
+#[derive(Debug)]
+pub(crate) struct Transports {
+    #[cfg(not(wasm_browser))]
+    ip: Vec<IpTransport>,
+    relay: Vec<RelayTransport>,
+}
+
+impl Transports {
+    /// Create a new transports structure.
```

Comment:
nit, especially since this is internal, but good habits and all that:
> The summary line should be written in third person singular present indicative form. Basically, this means write ‘Returns’ instead of ‘Return’.

https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text